## Final Project Submission

Please fill out:
* Student name: GEOFFREY MWANGI WACHIRA
* Student pace: full time
* Scheduled project review date/time: 03/05/2024
* Instructor name:Samuel G. Mwangi
* Blog post URL: N/A


# Business Understanding

## Objective
The purpose of this analysis is to help Microsoft understand which types of films are performing well at the box office, providing actionable insights for their new movie studio.

## Business Problem
Microsoft is looking to enter the movie industry but lacks knowledge about movie production and performance metrics. This analysis aims to uncover key trends in the movie industry to guide Microsoft's decisions on what types of films to produce.

## Goal
Our goal is to analyze movie data to provide three concrete business recommendations to help Microsoft make informed decisions about their new movie studio.


## Data
Here, we'll describe the datasets we'll be working with and their relevance to the business problem.

Data Source
s1. :
Box Office Mojo: Provides information on movie box office performanc2. e.
IMDB: Contains comprehensive data on movies, including ratings and cast information.
Rotten Tomatoes: Offers movie reviews and ratings from critics and audiences.
TheMovieDB: A database for movies and TV shows, providing detailed information on titles and crew.
The Numbers: Provides data on movie budgets, revenues, and production costs.Numbers]


## Project Outline
We will conduct EDA on the movie and movie rating databases to identify trends and patterns in box office success. We will analyze factors such as genre, time period, ratings, budget, director, actor, and production country to identify what makes a successfuland profitable  movie. We will also identify any seasonal trends in movie releases and revenue.rThis wil help us answer wer the following questions:

* What are the most successful genres at the box office?
* How do ratings and revenue vary across different genres and time periods?
* Are there any seasonal trends in movie releases and revenue?
* Who are the most successful directors and actors in terms of revenue and ratings?
* What is the average budget for successful movies, and how does it vary across different genres?
* Are there any relationships between critical acclaim (e.g. movie ratings from critics) and box office revenue?
* What are the most popular countries for movie production and distribution, and how do their box office revenues compare?
* Are there any patterns in the runtime of successful movies, and how do they vary across different genres?

**Importing necessary libraries**

In [2]:
import pandas as pd
import os 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
#Importing sqlite3 library
import sqlite3

# Data Preparation
In this section, we'll load and preprocess the datasets, ensuring they are ready for analysis

In [4]:
# Reading the data - Box Office Mojo - Movie Gross
bom_movie_gross = pd.read_csv('C:/Users/Hp/Desktop/Phase_One_Project/dsc-phase-1-project-v2-4/zippedData/bom.movie_gross.csv.gz')

# Connecting to the database
conn = sqlite3.connect('C:/Users/Hp/Desktop/Phase_One_Project/dsc-phase-1-project-v2-4/zippedData/im.db/im.db')
cur = conn.cursor()

# Loading the movie_info table
rt_movie_info = pd.read_csv('C:/Users/Hp/Desktop/Phase_One_Project/dsc-phase-1-project-v2-4/zippedData/rt.movie_info.tsv.gz', delimiter='\t')

# Loading rt.reviews table
rt_reviews = pd.read_csv('C:/Users/Hp/Desktop/Phase_One_Project/dsc-phase-1-project-v2-4/zippedData/rt.reviews.tsv.gz', delimiter='\t', encoding='latin-1')

# Loading tmdb data tables
tmdb_movies = pd.read_csv('C:/Users/Hp/Desktop/Phase_One_Project/dsc-phase-1-project-v2-4/zippedData/tmdb.movies.csv.gz')

# Loading tn.movie_budgets data table
tn_movie_budgets = pd.read_csv('C:/Users/Hp/Desktop/Phase_One_Project/dsc-phase-1-project-v2-4/zippedData/tn.movie_budgets.csv.gz')


Now, let's proceed with exploring each dataset in detail:
## 
1. Box Office Mojo - Movie Gross

In [19]:
# Shape
print("Box Office Mojo - Movie Gross's Shape is :", bom_movie_gross.shape)

print('************************************************************************')
print('************************************************************************')

# Displaying the first five elements of the dataset
print("First five elements:")
print(bom_movie_gross.head())

print('************************************************************************')
print('************************************************************************')

# Displaying the dataset info
print("Dataset info:")
bom_movie_gross.info()

print('************************************************************************')
print('************************************************************************')


# Descriptive statistics
print("Descriptive Statistics:")
print(bom_movie_gross.describe())

print('************************************************************************')
print('************************************************************************')

# Missing data
print("Missing Data:")
print(bom_movie_gross.isnull().sum())

print('************************************************************************')
print('************************************************************************')

# Duplicates
print("Duplicate Rows:")
print(bom_movie_gross[bom_movie_gross.duplicated()])

print('************************************************************************')
print('************************************************************************')


Box Office Mojo - Movie Gross's Shape is : (3387, 5)
************************************************************************
************************************************************************
First five elements:
                                         title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010  
************************************************************************
************************************************************************
Dataset info:
<class 'pandas.core.frame.DataFrame'>

### Conclusions:

The dataset contains 3387 entries and 5 columns.
The columns include 'title', 'studio', 'domestic_gross', 'foreign_gross', and 'year'.
The 'domestic_gross' and 'foreign_gross' columns contain numeric data, while 'studio' and 'title' are categorical.
There are missing values in the 'studio', 'domestic_gross', and 'foreign_gross' columns.
The 'foreign_gross' column is of object type instead of numeric, suggesting potential data formatting issues.
Descriptive statistics show a wide range of domestic gross revenues, with a mean of approximately $28.7 million.
The dataset does not contain any duplicate rows.8.


In [24]:

# Converting 'foreign_gross' to numeric
bom_movie_gross['foreign_gross'] = pd.to_numeric(bom_movie_gross['foreign_gross'], errors='coerce')

# Handling missing values
bom_movie_gross['domestic_gross'].fillna(bom_movie_gross['domestic_gross'].median(), inplace=True)
bom_movie_gross['foreign_gross'].fillna(0, inplace=True)
bom_movie_gross.dropna(subset=['studio'], inplace=True)


In [25]:
# Verifing changes
print(bom_movie_gross.info())
print(bom_movie_gross.isnull().sum())

<class 'pandas.core.frame.DataFrame'>
Index: 3382 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3382 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3382 non-null   float64
 3   foreign_gross   3382 non-null   float64
 4   year            3382 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 158.5+ KB
None
title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64


##2. Rotten Tomatoes - Movie Info

In [26]:
# Alook at the Shape
print("Rotten Tomatoes - Movie Info's Shape is :", rt_movie_info.shape)

print('************************************************************************')
print('************************************************************************')

# Displaying the first five elements of the dataset
print("First five elements:")
print(rt_movie_info.head())

print('************************************************************************')
print('************************************************************************')

# Displaying the dataset info
print("Dataset info:")
rt_movie_info.info()

print('************************************************************************')
print('************************************************************************') 

# Descriptive statistics
print("Descriptive Statistics:")
print(rt_movie_info.describe())

print('************************************************************************')
print('************************************************************************')

# Missing data
print("Missing Data:")
print(rt_movie_info.isnull().sum())

print('************************************************************************')
print('************************************************************************')

# Duplicates
print("Duplicate Rows:")
print(rt_movie_info[rt_movie_info.duplicated()])

print('************************************************************************')
print('************************************************************************')


Rotten Tomatoes - Movie Info's Shape is : (1560, 12)
************************************************************************
************************************************************************
First five elements:
   id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                

### Conclusion:
The dataset contains information about 1560 movies.
Several columns have missing values, including 'synopsis', 'rating', 'genre', 'director', 'writer', 'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime', and 'studio'.
The 'currency' and 'box_office' columns seem to have significant missing values (1220 out of 1560).
The 'runtime' column has a few missing values (30 out of 1560).
The 'studio' column has many missing values (1066 out of 1560).
The 'rating' column has only a few missing values (3 out of 1560).

In [27]:

# Converting 'box_office' to numeric
rt_movie_info['box_office'] = pd.to_numeric(rt_movie_info['box_office'], errors='coerce')

# Handling missing values
# Dropping 'currency' and 'box_office' columns
rt_movie_info.drop(columns=['currency', 'box_office'], inplace=True)  
# Dropping rows with missing values in selected columns
rt_movie_info.dropna(subset=['director', 'writer', 'theater_date', 'dvd_date', 'runtime'], inplace=True)
# Replacing missing 'synopsis' values with placeholder
rt_movie_info['synopsis'].fillna("No synopsis available", inplace=True)  
# Replacing missing 'genre' values with placeholder
rt_movie_info['genre'].fillna("Unknown", inplace=True) 
 # Replacing missing 'studio' values with placeholder
rt_movie_info['studio'].fillna("Unknown", inplace=True) 



In [ ]:
# Verifing changes
print(rt_movie_info.info())
print(rt_movie_info.isnull().sum())


## 3. Rotten Tomatoes - Reviews

In [28]:
# Shape
print("Shape:", rt_reviews.shape)
print('************************************************************************')
print('************************************************************************')

# Displaying the dataset info
print("Dataset info:")
rt_reviews.info()

print('************************************************************************')
print('************************************************************************')

# Displaying the first five elements of the dataset
print("First five elements:")
print(rt_reviews.head())

print('************************************************************************')
print('************************************************************************')

# Descriptive statistics
print("Descriptive Statistics:")
print(rt_reviews.describe())
print('************************************************************************')
print('************************************************************************')

# Missing data
print("Missing Data:")
print(rt_reviews.isnull().sum())
print('************************************************************************')
print('************************************************************************')

# Duplicates
print("Duplicate Rows:")
print(rt_reviews[rt_reviews.duplicated()])
print('************************************************************************')
print('************************************************************************')


Shape: (54432, 8)
************************************************************************
************************************************************************
Dataset info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB
************************************************************************
************************************************************************
First five elements:
   id                                             review rating   fresh  \
0   3  A distinctly ga

### Conclusion:
The dataset contains information about 54432 movie reviews.
Columns like 'review', 'rating', 'critic', and 'publisher' have missing values.
The 'rating' column contains ratings in different formats (e.g., 3/5, 4.5/5, 4/5, 2/5, 0.5/5), which might need standardization.
Some duplicate rows are present in the dataset.

In [34]:

# Handling missing values
rt_reviews['review'].fillna("No review available", inplace=True)  # Replace missing 'review' values with placeholder
rt_reviews['critic'].fillna("Unknown", inplace=True)  # Replace missing 'critic' values with placeholder
rt_reviews['publisher'].fillna("Unknown", inplace=True)  # Replace missing 'publisher' values with placeholder
rt_reviews.dropna(subset=['rating'], inplace=True)  # Drop rows with missing 'rating' values

# Converting 'rating' column to string
rt_reviews['rating'] = rt_reviews['rating'].astype(str)

# Removing '/5' from ratings
rt_reviews['rating'] = rt_reviews['rating'].str.replace('/5', '')

# Converting ratings to numeric format
rt_reviews['rating'] = pd.to_numeric(rt_reviews['rating'], errors='coerce')

# Dropping rows with missing 'rating' values
rt_reviews.dropna(subset=['rating'], inplace=True)


# Removing duplicate rows
rt_reviews.drop_duplicates(inplace=True)


In [36]:
# Verifying changes
print(rt_reviews.info())
print(rt_reviews.isnull().sum())

<class 'pandas.core.frame.DataFrame'>
Index: 17785 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          17785 non-null  int64  
 1   review      17785 non-null  object 
 2   rating      17785 non-null  float64
 3   fresh       17785 non-null  object 
 4   critic      17785 non-null  object 
 5   top_critic  17785 non-null  int64  
 6   publisher   17785 non-null  object 
 7   date        17785 non-null  object 
dtypes: float64(1), int64(2), object(5)
memory usage: 1.2+ MB
None
id            0
review        0
rating        0
fresh         0
critic        0
top_critic    0
publisher     0
date          0
dtype: int64


4. The Movie Database (TMDB) - Movies

In [40]:
# Shape
print("Shape:", tmdb_movies.shape)

print('************************************************************************')
print('************************************************************************')

# Displaying the dataset info
print("Dataset info:")
tmdb_movies.info()

print('************************************************************************')
print('************************************************************************')

print("First five elements:")
print(tmdb_movies.head())

print('************************************************************************')
print('************************************************************************')

# Descriptive statistics
print("Descriptive Statistics:")
print(tmdb_movies.describe())

print('************************************************************************')
print('************************************************************************')

# Missing data
print("Missing Data:")
print(tmdb_movies.isnull().sum())

print('************************************************************************')
print('************************************************************************')

# Duplicates
print("Duplicate Rows:")
print(tmdb_movies[tmdb_movies.duplicated()])

print('************************************************************************')
print('************************************************************************')


Shape: (26517, 8)
************************************************************************
************************************************************************
Dataset info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   id                 26517 non-null  int64         
 1   original_language  26517 non-null  object        
 2   original_title     26517 non-null  object        
 3   popularity         26517 non-null  float64       
 4   release_date       26517 non-null  datetime64[ns]
 5   title              26517 non-null  object        
 6   vote_average       26517 non-null  float64       
 7   vote_count         26517 non-null  int64         
dtypes: datetime64[ns](1), float64(2), int64(2), object(3)
memory usage: 1.6+ MB
************************************************************************
**********

### Conclusions

There are no missing values in any of the columns.
No duplicate rows were found in the dataset
The 'release_date' column is in string format and should be converted to datetime for further analysis..

In [41]:
# Converting 'release_date' to datetime format
tmdb_movies['release_date'] = pd.to_datetime(tmdb_movies['release_date'])

# Droping unnecessary columns
tmdb_movies.drop(columns=['Unnamed: 0', 'genre_ids'], inplace=True)


KeyError: "['Unnamed: 0', 'genre_ids'] not found in axis"

In [42]:
# Verifying changes
print(tmdb_movies.info())
print(tmdb_movies.isnull().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   id                 26517 non-null  int64         
 1   original_language  26517 non-null  object        
 2   original_title     26517 non-null  object        
 3   popularity         26517 non-null  float64       
 4   release_date       26517 non-null  datetime64[ns]
 5   title              26517 non-null  object        
 6   vote_average       26517 non-null  float64       
 7   vote_count         26517 non-null  int64         
dtypes: datetime64[ns](1), float64(2), int64(2), object(3)
memory usage: 1.6+ MB
None
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64


5. The Numbers - Movie Budgets

In [43]:
# Shape
print("Shape:", tn_movie_budgets.shape)

print('************************************************************************')
print('************************************************************************')

# Displaying the dataset info
print("Dataset info:")
tn_movie_budgets.info()

print('************************************************************************')
print('************************************************************************')

print("First five elements:")
print(tn_movie_budgets.head())

print('************************************************************************')
print('************************************************************************')

# Descriptive statistics
print("Descriptive Statistics:")
print(tn_movie_budgets.describe())

print('************************************************************************')
print('************************************************************************')

# Missing data
print("Missing Data:")
print(tn_movie_budgets.isnull().sum())

print('************************************************************************')
print('************************************************************************')

# Duplicates
print("Duplicate Rows:")
print(tn_movie_budgets[tn_movie_budgets.duplicated()])

print('************************************************************************')
print('************************************************************************')


Shape: (5782, 6)
************************************************************************
************************************************************************
Dataset info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB
************************************************************************
************************************************************************
First five elements:
   id  release_date                                        movie  \
0   1  Dec 18, 2009                                   

### Conclusions

There are no missing values in any of the columns.
No duplicate rows were found in the dataset.

The dataset contains information about movie releases, including movie titles, release dates, production budgets, domestic and worldwide gross revenues.

The 'release_date' column is in string format and should be converted to datetime

The 'production_budget', 'domestic_gross', and 'worldwide_gross' columns are currently in string format and should be converted to numeric

The 'production_budget' column represents the budget allocated for producing the movie, which can be compared with the gross revenue to assess profitability.

In [45]:
# Converting 'release_date' to datetime format
tn_movie_budgets['release_date'] = pd.to_datetime(tn_movie_budgets['release_date'])

# Converting budget and gross columns to numeric format
tn_movie_budgets['production_budget'] = tn_movie_budgets['production_budget'].str.replace('$', '').str.replace(',', '').astype(float)
tn_movie_budgets['domestic_gross'] = tn_movie_budgets['domestic_gross'].str.replace('$', '').str.replace(',', '').astype(float)
tn_movie_budgets['worldwide_gross'] = tn_movie_budgets['worldwide_gross'].str.replace('$', '').str.replace(',', '').astype(float)


In [47]:
# Confirming the changes
tn_movie_budgets.head()

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0        425000000.0     760507625.0     2.776345e+09  
1        410600000.0     241063875.0     1.045664e+09  
2        350000000.0      42762350.0     1.497624e+08  
3        330600000.0     459005868.0     1.403014e+09  
4        317000000.0     620181382.0     1.316722e+09